src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ILGenerator.Mono.cs

```diff
@@ -887,6 +888,7 @@ public virtual void EmitCalli(OpCode opcode, CallingConventions callingConventio
             Emit(opcode, helper);
         }
 
+        [return: DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
```

Comment:
Why all public methods when we need only `WriteLine` ?

Reply:
Because it's the safe thing to do - the other option would be to place DynamicDependency and suppress warnings at places that use this, but DynamicDependency along with warning suppression opens things up to human error and I wouldn't ever use DynamicDependency without dedicated test coverage.

The thing reflected on is going to change if/when Mono aligns with CoreCLR (CoreCLR doesn't call WriteLine on Console, but calls get_Out first and calls WriteLine on that, for whatever reason - the reflected thing is different and it's too easy to miss that).

Reply:
We could also just inline this helper and nothing needs to be annotated...

Reply:
Sounds better to me